src/neo-vm/ExecutionEngine.cs

```diff
@@ -1207,6 +1266,55 @@ private bool ExecuteStoreToSlot(Slot slot, int index)
             return true;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private bool ExecuteTry(int catchOffset, int finallyOffset)
+        {
+            if (catchOffset == 0 && finallyOffset == 0) return false;
+            int catchPointer = catchOffset == 0 ? -1 : checked(CurrentContext.InstructionPointer + catchOffset);
+            int finallyPointer = finallyOffset == 0 ? -1 : checked(CurrentContext.InstructionPointer + finallyOffset);
+            CurrentContext.TryStack ??= new Stack<ExceptionHandingContext>();
+            CurrentContext.TryStack.Push(new ExceptionHandingContext(catchPointer, finallyPointer));
+            return true;
+        }
+
+        private bool HandleException()
+        {
+            int pop = 0;
+            foreach (var executionContext in InvocationStack)
+            {
+                if (executionContext.TryStack != null)
+                {
+                    while (executionContext.TryStack.TryPeek(out var tryContext))
+                    {
+                        if (tryContext.State == ExceptionHandingState.Finally || (tryContext.State == ExceptionHandingState.Catch && !tryContext.HasFinally))
+                        {
+                            executionContext.TryStack.Pop();
+                            continue;
+                        }
+                        for (int i = 0; i < pop; i++)
+                        {
+                            ContextUnloaded(InvocationStack.Pop());
```

Comment:
We are changing a collection (`InvocationStack`) inside the foreach loop. Are you sure that this works?

Reply:
This lines are not tested 
https://coveralls.io/builds/30008902/source?filename=src/neo-vm/ExecutionEngine.cs#L1294

Reply:
Then the method returns.

Reply:
Done, I have added the ut, it worked well.